### PR TITLE
Update 2022-12-12-async-manual.md

### PR DESCRIPTION
**Pull request by parttimenerd**

### CODE REVIEW

File: _posts/2022-12-12-async-manual.md [hunk 1]
```diff
@@ -740,74 +735,69 @@ public static int[][] matrixMultiplyFaster(int[][] a, int[][] b, int size) {
...
 
 - Knowledge of CPU architecture
 - Top-Down performance analysis methodology
 - Looking at the ASM of generated methods  
 
-Many Java programmers forget that all the execution is done in CPU. To talk to CPU Java needs to use ASM. That's
-basically what the JIT compiler is doing, it converts your hot methods and loops into effective ASM. At the assembly level
-you can check if JIT used vectorized instruction for your loops for example. So yes, sometimes you need to get dirty 
-with such low level stuff. For now async-profiler can give you a hint which methods you should focus on.
+Many Java programmers forget that all the execution is done on the CPU. Java needs to use tak ASM to run on the CPU. That's
```
Comment (krzysztofslusarski) [inline _posts/2022-12-12-async-manual.md]:
```suggestion
Many Java programmers forget that all the execution is done on the CPU. Java needs to use ASM to run on the CPU. That's
```

File: _posts/2022-12-12-async-manual.md [hunk 2]
```diff
@@ -740,74 +735,69 @@ public static int[][] matrixMultiplyFaster(int[][] a, int[][] b, int size) {
...
 
-Introducing TLABs creates two more issues that JVM needs to deal:
+Introducing TLABs creates two more issues that the JVM needs to deal with:
 
-- a thread can allocate an object, but there is not enough space in its TLAB - in that case
-  if there is still a space in eden JVM creates new TLAB for that thread
-- a thread can allocate a big object, so it's not optimal to use TLAB mechanism - in that case
-  JVM will use _slow path_ of allocation that allocates the object directly in eden
+- a thread can allocate an object, but there is not enough space in its TLAB - the JVM creates a new TLAB if there is still space in eden
+- a thread can allocate a big object, so it's not optimal to use the TLAB mechanism - the JVM will use the _slow path_ of the allocation that allocates the object directly in eden
```
Comment (krzysztofslusarski) [inline _posts/2022-12-12-async-manual.md]:
```suggestion
- a thread can allocate a big object, so it's not optimal to use the TLAB mechanism - the JVM will use the _slow path_ of the allocation that allocates the object directly in eden or in the old generation
```

Comment (krzysztofslusarski) [inline _posts/2022-12-12-async-manual.md]: G1 Humongous objects are created directly in the old-gen, forgot about it before.

File: _posts/2022-12-12-async-manual.md [hunk 3]
```diff
@@ -983,52 +972,57 @@ GC(11536) Pause Young (Normal) (G1 Evacuation Pause) 6746M->2016M(8192M) 40.514m
...
 ![alt text](/assets/monday-2/4.jpg "4")
 
-then there is **no memory leak**. The _garbage collector_ can clean up the heap to the same level every day. The chart with memory leak looks like this one:
+then there is **no memory leak**. The _garbage collector_ can clean up the heap to 
+the same level every day. The chart with a memory leak looks like the following one:
 
 ![alt text](/assets/monday-2/2.jpg "2")
 
-These spikes to the roof are _to-space exhausted_ situations in the **G1** algorithm, those are not _OutOfMemoryErrors_. After each of those spikes there was
+These spikes to the roof are _too-space exhausted_ situations in the **G1** algorithm; those are not _OutOfMemoryErrors_. After each of those spikes, there was a
```
Comment (krzysztofslusarski) [inline _posts/2022-12-12-async-manual.md]:
```suggestion
These spikes to the roof are _to-space exhausted_ situations in the **G1** algorithm; those are not _OutOfMemoryErrors_. After each of those spikes, there was a
```

File: _posts/2022-12-12-async-manual.md [hunk 4]
```diff
@@ -1462,17 +1453,17 @@ protected Exception(String message, Throwable cause,
...
 {: #methods-g1ha }
 
-I've already mentioned how to detect humongous objects with allocation mode. Since an
-async-profiler can also instrument JVM code, and allocation of a humongous object is 
-nothing else than invocation of part of a code in C, we can take advantage of that.
-If you want to check where humongous objects are allocated you can use method mode
+We already saw how to detect humongous objects with allocation mode. Since
+async-profiler can also instrument JVM code, and allocations of a humongous objects are 
+nothing else than invocations of C++ code, we can take advantage of that.
+If you want to check where humongous objects are allocated, you can use method mode
```
Comment (krzysztofslusarski) [inline _posts/2022-12-12-async-manual.md]:
```suggestion
If you want to check where humongous objects are allocated, you can use native functions mode
```

Comment (krzysztofslusarski) [reply _posts/2022-12-12-async-manual.md]: Method mode is not the same as native functions. It's a bit undocumented, but Andrei ask me to separate those two.

File: _posts/2022-12-12-async-manual.md [hunk 5]
```diff
@@ -1506,8 +1496,8 @@ If you monitor that value and the chart like that:
 
 ![alt text](/assets/async-demos/threads-2.png "threads")
 
-Then you may check who is creating those short living threads. You can find out that
-with method the mode of async-profiler using the ```JVM_StartThread``` event.
+Then you might want to check who is creating those short-living threads:
+We use async-profiler with the ```JVM_StartThread``` event in method mode for this purpose:
```
Comment (krzysztofslusarski) [inline _posts/2022-12-12-async-manual.md]:
```suggestion
We use async-profiler with the ```JVM_StartThread``` event in native functions mode for this purpose:
```

File: _posts/2022-12-12-async-manual.md (hunk 4 above)
Comment (parttimenerd) [reply _posts/2022-12-12-async-manual.md]: Good to know

**Pull request merged**